src/views/Home.jsx

```diff
@@ -1,12 +1,52 @@
 import './Home.css';
 import { SingleList } from '../components/SingleList';
+import { useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { createList } from '../api/firebase.js';
+import { useAuth } from '../api';
 
 export function Home({ data, setListPath }) {
+	const { user } = useAuth();
+	const userId = user?.uid;
+	const userEmail = user?.email;
+	const navigate = useNavigate();
+
+	const [listName, setListName] = useState('');
+
+	const handleInputChange = (event) => {
+		setListName(event.target.value);
+	};
+
+	const handleSubmit = async (event) => {
+		event.preventDefault();
+		if (listName) {
+			const path = await createList(userId, userEmail, listName);
+			setListPath(path);
+			setListName('');
+			navigate('/list');
```

Comment:
Yes! love that you are redirecting based on the submit event vs wiring up a `useEffect` - this is definitely best practice, great job getting there!